superset/connectors/base/models.py

```diff
@@ -317,11 +317,23 @@ def data_for_slices(self, slices: List[Slice]) -> Dict[str, Any]:
                 if "column" in filter_config
             )
 
-            column_names.update(
-                column
-                for column_param in COLUMN_FORM_DATA_PARAMS
-                for column in utils.get_iterable(form_data.get(column_param) or [])
-            )
+            # legacy charts don't have query_context charts
+            if slc.query_context:
+                query_context = slc.get_query_context()
+                column_names.update(
+                    [
+                        column
+                        for query in query_context.get("queries", [])
+                        for column in query.get("columns", [])
+                    ]
+                    or []
+                )
+            else:
+                column_names.update(
+                    column
+                    for column_param in COLUMN_FORM_DATA_PARAMS
+                    for column in utils.get_iterable(form_data.get(column_param) or [])
+                )
```

Comment:
I think just add `groupbyColumns` and `groupbyRows` into COLUMN_FORM_DATA_PARAMS

````
COLUMN_FORM_DATA_PARAMS = [
    "all_columns",
    "all_columns_x",
    "columns",
    "entity",
    "groupby",
    "order_by_cols",
    "series",
    "groupbyColumns",
    "groupbyRows",
]
```

Reply:
That was actually my first solution 😄 But @villebro suggested that I use columns in query_context instead, so that the solution is more generic and future proof - if in the future we add another chart that uses different field names in form data for columns, this solution will handle it correctly

Reply:
Thanks for the explanation. I understood.